MAUI/DataGrid/columns.md

```diff
@@ -0,0 +1,205 @@
+</tr>
+<tr>
+<td> <code>Reset</code> </td>
+<td> Retains the columns defined explicitly in the application level and creates columns newly for all the other properties in a data source.<br/> When changing the <code>ItemsSource</code> and sorting for explicitly defined <code>SfDataGrid.Columns</code> alone will be retained. </td>
+</tr>
+<tr>
+<td> <code>ResetAll</code> </td>
+<td> When changing the <code>ItemsSource</code>, the columns for the previous data source are cleared and the columns will be created newly for the new data source. Even when columns are explicitly defined it does not consider the defined columns and creates the column based on the underlying collection.<br/> Further when changing the <code>ItemsSource</code> and sorting for all the columns will be cleared. </td>
+</tr>
+<tr>
+<td> <code>RetainOld</code> </td>
+<td> When changing the <code>ItemsSource</code>, creates columns for all fields in a data source when the DataGrid does not have any explicit definition for columns. When columns are defined explicitly, then the defined columns alone are retained and new columns are not created.<br/>Similarly when changing the <code>ItemsSource</code> and when the DataGrid have any explicit definition for columns and sorting are retained as it is. </td>
+</tr>
+<tr>
+<td> <code>SmartReset</code> </td>
+<td> Retains the columns defined explicitly in application level and the columns with <code>MappingName</code> identical to the properties in the new data source. Creates columns newly for all the other properties in the data source.<br/> Similarly it retains the sorting of the columns that are defined explicitly in application level and the columns with <code>MappingName</code> identical to the properties in new data source. </td>
+</tr>
+</table>
+
+### Customize automatically generated columns
+
+The auto generated column can be customized by handling the [SfDataGrid.AutoGeneratingColumn](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_AutoGeneratingColumn) event. The event will be invoked when the column is auto-generated.
+
+The [DataGridAutoGeneratingColumnEventArgs](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridAutoGeneratingColumnEventArgs.html) object contains the following properties,
+
+* **Column**: This property returns the created column which can be customized.
+* **Cancel**: This property cancels the column creation.
+
+You can skip generating a column by handling the `SfDataGrid.AutoGeneratingColumn` event as shown as follows,
+
+{% tabs %}
+{% highlight c# %}
+this.sfDataGrid.AutoGeneratingColumn += SfDataGrid_AutoGeneratingColumn;
+
+private void SfDataGrid_AutoGeneratingColumn(object sender, DataGridAutoGeneratingColumnEventArgs e)
+{
+	if(e.Column.MappingName == "ShipCountry")
+	{
+		e.Cancel = true;
+	}
+}
+{% endhighlight %}
+{% endtabs %}
+
+Formatting for the auto generated columns can be applied as follows:

```

Comment:
Formatting for the auto-generated columns can be applied as follows:

Reply:
Modified.